src/holdup/cli.py

```diff
@@ -237,6 +256,14 @@ def parse_value(value, proto):
             raise argparse.ArgumentTypeError('Invalid service spec %r. Port must be a number.' % value)
         port = int(port)
         return TcpCheck(host, port)
+    elif proto in ('pg', 'postgresql', 'postgres'):
+        uri = 'postgresql://{}'.format(value)
+        try:
+            connection_uri = make_dsn(uri)
+        except ProgrammingError:
```

Comment:
Should capture all exceptions (`except Exception as exc:`) and `raise argparse.ArgumentTypeError("Failed to parse %s: %s" % (value, exc))` or similar.